### ISSUE TITLE
How to run?

### ISSUE THREAD

**Issue by MrNunoski** [nikhil5bhai/ShopStyle#1]
Hi Nikhil,

Question - How do I run your code?

Trying to run from command line..  "mvn test" produces the following errors:

[INFO] No sources to compile
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ ShopStyle-APITest-001 ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory /Users/ngoncalves/workspace/interviews/ShopStyle/ShopStyle-APITest-001/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ ShopStyle-APITest-001 ---
[INFO] Changes detected - recompiling the module!
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] Compiling 1 source file to /Users/ngoncalves/workspace/interviews/ShopStyle/ShopStyle-APITest-001/target/test-classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/ngoncalves/workspace/interviews/ShopStyle/ShopStyle-APITest-001/src/test/java/com/ShopStype/tests/ShopStyleGet.java:[13,17] package org.junit does not exist
[ERROR] /Users/ngoncalves/workspace/interviews/ShopStyle/ShopStyle-APITest-001/src/test/java/com/ShopStype/tests/ShopStyleGet.java:[21,10] cannot find symbol
  symbol:   class Test
  location: class com.ShopStype.tests.ShopStyleGet
[ERROR] /Users/ngoncalves/workspace/interviews/ShopStyle/ShopStyle-APITest-001/src/test/java/com/ShopStype/tests/ShopStyleGet.java:[45,10] cannot find symbol
  symbol:   class Test
  location: class com.ShopStype.tests.ShopStyleGet
[INFO] 3 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 9.589 s
[INFO] Finished at: 2017-06-06T17:03:21-07:00
[INFO] Final Memory: 15M/245M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:testCompile (default-testCompile) on project ShopStyle-APITest-001: Compilation failure: Compilation failure:
[ERROR] /Users/ngoncalves/workspace/interviews/ShopStyle/ShopStyle-APITest-001/src/test/java/com/ShopStype/tests/ShopStyleGet.java:[13,17] package org.junit does not exist
[ERROR] /Users/ngoncalves/workspace/interviews/ShopStyle/ShopStyle-APITest-001/src/test/java/com/ShopStype/tests/ShopStyleGet.java:[21,10] cannot find symbol
[ERROR] symbol:   class Test
[ERROR] location: class com.ShopStype.tests.ShopStyleGet
[ERROR] /Users/ngoncalves/workspace/interviews/ShopStyle/ShopStyle-APITest-001/src/test/java/com/ShopStype/tests/ShopStyleGet.java:[45,10] cannot find symbol
[ERROR] symbol:   class Test
[ERROR] location: class com.ShopStype.tests.ShopStyleGet
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

**Comment by nikhil5bhai** [comment]
Hi,

I have updated the repo with minor dependency change. Please check it

**Comment by MrNunoski** [comment]
How do I run this? I am not using an IDE I am running from the command line.

**Comment by nikhil5bhai** [comment]
If you are using Windows, confirm that -
1. Java is installed and environment variables are set
2. Maven installed - Environment variables are properly set
3. Save the project on your machine
4. Open the command prompt (Go inside the folder containing pom.xml). Follow demo -https://www.youtube.com/watch?v=jaaYHCKtWmA
3. Run this command -  mvn clean verify -Dtest=ShopStyleGet

//Presently no idea about the Mac, but try the same in terminal

**Comment by MrNunoski** [comment]
Got this, but doesn't seem like you finished solving the problem.. You only pulled maximum of 10 pages..

 * As long as the "more" field returns true, you have more data available.
 * The response is a JSON object which has a response key which is an array of more JSON objects. 
 * Each of them has a key called flags and within flags there is a key called hd.
 * Print out how many response objects have flags:hd set to true and how many have hd set to false.

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.ShopStype.tests.ShopStyleGet
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=1
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=2
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=4
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=5
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=6
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=7
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=8
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=9
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=10
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=1
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=2
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=4
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=5
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=6
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=7
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=8
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=9
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=10
Total HD False Count = 0
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.956 sec

Results :

Tests run: 2, Failures: 0, Errors: 0, Skipped: 0

[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ ShopStyle-APITest-001 ---
[WARNING] JAR will be empty - no content was marked for inclusion!
[INFO] Building jar: /Users/ngoncalves/workspace/interviews/ShopStyle/ShopStyle-APITest-001/target/ShopStyle-APITest-001-0.0.1-SNAPSHOT.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 7.199 s
[INFO] Finished at: 2017-06-12T09:57:36-07:00
[INFO] Final Memory: 19M/245M
[INFO] ------------------------------------------------------------------------

**Comment by nikhil5bhai** [comment]
Right. It's my fault in understanding the question in my way. As the no. of api(s) can't be infinite, earlier I tried using different small ranges till 2000 and the result was almost the same.  Thats why I kept the max no pages - j in code with comment as - 
int jmax = 10; // jmax is maximum number of page starting from 1 which can be edited by user

I just tried using more ranges and got the result as- 
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3500
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3501
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3502
Total HD True Count = 8
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3503
Total HD True Count = 7
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3504
Total HD True Count = 8
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3505
Total HD True Count = 8
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3506
Total HD True Count = 8
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3507
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3508
Total HD True Count = 3
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3509
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3510
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3511
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3512
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3513
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3514
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3515
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3516
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3517
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3518
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3519
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3520
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3521
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3522
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3523
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3524
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3525
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3526
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3527
Total HD True Count = 0

The value of "more" field is found to be 3508 if no more values for pages are expected.

**Comment by nikhil5bhai** [comment]
Right. It's my mistake in understanding the question in my way. As the no. of api(s) can't be infinite, earlier I tried using different small ranges till 2000 and the result was almost the same. Thats why I kept the max no pages - j in code with comment as -
int jmax = 10; // jmax is maximum number of page starting from 1 which can be edited by user
I just tried using more ranges and got the result as-
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3500
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3501
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3502
Total HD True Count = 8
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3503
Total HD True Count = 7
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3504
Total HD True Count = 8
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3505
Total HD True Count = 8
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3506
Total HD True Count = 8
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3507
Total HD True Count = 10
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3508
Total HD True Count = 3
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3509
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3510
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3511
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3512
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3513
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3514
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3515
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3516
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3517
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3518
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3519
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3520
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3521
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3522
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3523
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3524
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3525
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3526
Total HD True Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3527
Total HD True Count = 0

The value of "more" field for fd:true is found to be 3508 if no more values for pages are expected.

And the result for hf:false is non consistent-

http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3450
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3451
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3452
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3453
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3454
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3455
Total HD False Count = 0 _____________________________________
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3470
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3471
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3472
Total HD False Count = 2
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3473
Total HD False Count = 2
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3474
Total HD False Count = 1
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3475
Total HD False Count = 1_____________________________________________
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3501
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3502
Total HD False Count = 2
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3503
Total HD False Count = 3
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3504
Total HD False Count = 2
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3505
Total HD False Count = 2
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3506
Total HD False Count = 2
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3507
Total HD False Count = 0
http://api.viki.io/v4/videos.json?app=100250a&per_page=10&page=3508
Total HD False Count = 0

(This output was tested on Eclipse IDE)

**Comment by nikhil5bhai** [comment]
I would be very thankful to you if you provide me the correct expected code for my learning purpose. Thanks.

**Comment by MrNunoski** [comment]
Hi Nikhil.  Sure, appreciate you reaching out again.

The way you're checking for the hd flag and incrementing is correct.

Your original code - instead of being junit tests, should have just been a class with methods to calculate the solution (all hd true objects in the entire page catalog).  The delivery was a bit awkward in being junit tests that only do a few of the page requests. JUnit is a good tool to know, but it should also be used to do it's main intention- which is testing. In your tests, for example, there are not any assertions that do any checks. In both you just do the calculation and then print the result.

Instead, write methods that solve for one pages at a time (or many), and then use JUnit to test your method(s) with assertions on expected results for each input you pass in. A solution like this actually goes beyond the problem which was asked.

One more bit of feedback: https://github.com/nikhil5bhai/ShopStyle
You have no Readme file, so anyone coming to your project doesn't have any knowledge of running your program.

Hope this helps.

**Comment by nikhil5bhai** [comment]
Thanks for your reply and guidance :-)
